src/server/pps/server/api_server.go

```diff
@@ -328,6 +355,30 @@ func (a *apiServer) DeleteJob(ctx context.Context, request *pps.DeleteJobRequest
 	return &types.Empty{}, nil
 }
 
+func (a *apiServer) RestartDatum(ctx context.Context, request *pps.RestartDatumRequest) (response *types.Empty, retErr error) {
+	func() { a.Log(request, nil, nil, 0) }()
+	defer func(start time.Time) { a.Log(request, response, retErr, time.Since(start)) }(time.Now())
+	metricsFn := metrics.ReportUserAction(ctx, a.reporter, "DeleteJob")
+	defer func(start time.Time) { metricsFn(start, retErr) }(time.Now())
+
+	jobInfo, err := a.InspectJob(ctx, &pps.InspectJobRequest{
+		Job: request.Job,
+	})
+	if err != nil {
+		return nil, err
+	}
+	var wp WorkerPool
+	if jobInfo.Pipeline != nil {
+		wp = a.workerPool(ctx, PipelineRcName(jobInfo.Pipeline.Name, jobInfo.PipelineVersion), RetriesPerDatum)
```

Comment:
If I understand the code correctly, there might be a leak here.  Before, there's this invariant that a `workerPool` is always tied to a pipeline or an orphaned job.  When the pipeline/job terminates, the worker pool (along with the goros it spawns) is deleted by means of context cancellation.

Now, it seems like every call to `WorkerStatus` or `RestartDatum` can create a worker pool, which in turn spawns a number of worker goros.  Are those goros being terminated somehow?  If not, that could be a leak.